### PR TITLE
Add unit test for zero fee transaction.

### PR DESCRIPTION
**Pull request by scottschurr**
Some time ago there was a bit of thrashing around regarding the lone zero-fee transaction we support: creating the first regular key on an account.  We determined that you could only attach the regular key for free if you signed with the master key.  You can't attach a regular key to an account for free if you multisign.

Later I realized that we did not have that scenario covered in the unit tests.  Now it will be.

Reviewers: @ximinez, @nbougalis

### CODE REVIEW

File: src/ripple/app/tests/MultiSign.test.cpp [hunk 1]
```diff
@@ -621,6 +621,39 @@ class MultiSign_test : public beast::unit_test::suite
...
+    // master key, but not when multisigning.
+    void test_regKey()
+    {
+        using namespace jtx;
+        Env env(*this);
+        Account const alice {"alice", KeyType::secp256k1};
+        env.fund(XRP(1000), alice);
+
+        // Give alice a regular key with a zero fee.  Should succeed.  Once.
+        Account const alie {"alie", KeyType::ed25519};
```
Comment (ximinez) [inline src/ripple/app/tests/MultiSign.test.cpp]: Tests generally don't need to specify they keytype unless you are testing something specific to that key type. It doesn't hurt, but it doesn't really add anything to the test either.

**Comment by ximinez** [comment]
Good test. :+1:

File: src/ripple/app/tests/MultiSign.test.cpp (hunk 1 above)
Comment (scottschurr) [inline src/ripple/app/tests/MultiSign.test.cpp]: Right.  I got in the habit of varying the key types in the multisign tests because the new key type was introduced in the middle of multisign development.  I wanted to have some confidence that things would continue to work with heterogeneous key types.  I'm not worried about specific key types here, I'm just continuing the model that's used elsewhere in the file.  As you say, it doesn't hurt.

**Comment by nbougalis** [comment]
:+1:

**Comment by nbougalis** [comment]
Merged as a8cc9033b49be348acf6053f01ed924703de344e.
